llrt_core/src/modules/http/headers.rs

```diff
@@ -1,24 +1,47 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
-use std::collections::BTreeMap;
+use std::{collections::BTreeMap, fmt};
 
 use hyper::HeaderMap;
 use rquickjs::{
-    atom::PredefinedAtom, methods, prelude::Opt, Array, Coerced, Ctx, FromJs, Function, Result,
-    Value,
+    atom::PredefinedAtom, methods, prelude::Opt, Array, Coerced, Ctx, FromJs, Function, IntoJs,
+    Result, Value,
 };
 
-use crate::utils::{
-    class::IteratorDef,
-    object::{array_to_btree_map, map_to_entries},
-};
+use crate::utils::{class::IteratorDef, object::array_to_btree_map};
+
+const HEADERS_KEY_SET_COOKIE: &str = "set-cookie";
+
+#[derive(Clone)]
+pub enum HeaderValue {
+    Single(String),
+    Multiple(Vec<String>),
+}
+
+impl fmt::Display for HeaderValue {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            HeaderValue::Single(s) => write!(f, "{}", s),
+            HeaderValue::Multiple(v) => write!(f, "{}", v.join("")),
+        }
+    }
+}
+
+impl<'js> IntoJs<'js> for HeaderValue {
+    fn into_js(self, ctx: &Ctx<'js>) -> Result<Value<'js>> {
+        match self {
+            HeaderValue::Single(s) => s.into_js(ctx),
+            HeaderValue::Multiple(v) => v.join(", ").into_js(ctx),
+        }
+    }
+}
 
 #[derive(Clone, Default)]
 #[rquickjs::class]
 #[derive(rquickjs::class::Trace)]
 pub struct Headers {
     #[qjs(skip_trace)]
-    headers: BTreeMap<String, String>,
+    headers: BTreeMap<String, HeaderValue>,
```

Comment:
> We can convert to Vec instead. That way we keep insertion order and is likely more performant for lower quantity of items < 1000. I see that Node uses a separate Vec for keeping track of `set-cookie` values. 

I may not understand the fine wording in English, so let me check.
`Node` refers to `Node.js`, and it is said that `Node.js` has an array to manage `set-cookie` separately from regular headers?

> Not sure if that's a good approach here or if we should keep the enum or simply use a single vec and allow for duplicate set-cookie values.

Since you raised the issue, you have been concerned about how it should be implemented...
Are there any of these that you would really like to have?

```rust
// No.1
pub struct Headers {
    headers: BTreeMap<String, String>
}
// However, only `set-cookie` is managed by `Vec<(String,String)>` etc.
```
```rust
// No.2
pub struct Headers {
    headers: Vec<(String,String)>
}
```
```rust
// No.3
pub struct Headers {
    headers: BTreeMap<String, Vec<String>>
}
```

Reply:
I would say 2 👌